File: src/main/java/site/shamota/backend/todo/service/StatService.java
package site.shamota.backend.todo.service;

import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;
import site.shamota.backend.todo.entity.Stat;
import site.shamota.backend.todo.repo.StatRepository;

@Service
@Transactional
public class StatService {

    private final StatRepository repository;

    public StatService(StatRepository repository) {
        this.repository = repository;
    }

    public Stat findStat(String email) {
        return repository.findByUserEmail(email);
    }
}

File: src/main/java/site/shamota/backend/todo/service/StatService.kt
package site.shamota.backend.todo.service

import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional
import site.shamota.backend.todo.repo.StatRepository
import site.shamota.backend.todo.entity.Stat

@Service
@Transactional
class StatService(private val repository: StatRepository) {
    fun findStat(email: String): Stat {
        return repository.findByUserEmail(email)
    }
}